content/ko/docs/concepts/architecture/controller.md

```diff
@@ -0,0 +1,162 @@
+---
+title: 컨트롤러
+content_template: templates/concept
+weight: 30
+---
+
+{{% capture overview %}}
+
+로봇과 자동화에서 _컨트롤 루프_ 는 
+시스템 상태를 조절하는 종료되지 않는 루프이다.
+
+여기 컨트롤 루프의 한 예시가 있다: 실내 온도 조절기.
+
+사용자가 온도를 설정하면, 온도 조절기는 사용자가 
+*의도한 상태* 를 알려준다. 실제 실내 온도는 
+*현재 상태* 이다. 온도 조절기는 장비를 작동하거나 종료해서 
+현재 상태를 의도한 상태에 가깝게 만든다.
+
+{{< glossary_definition term_id="controller" length="short">}}
+
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+## 컨트롤러 패턴
+
+컨트롤러는 적어도 하나 이상의 쿠버네티스 리소스 유형을 추적한다.
+이 [오브젝트](/ko/docs/concepts/overview/working-with-objects/kubernetes-objects/)
+는 의도한 상태를 표현하는 사양 필드를 가지고 있다.
+해당 리소스의 컨트롤러(들)은 현재 상태를 의도한 
+상태에 가깝게 만드는 역할을 한다.
+
+컨트롤러는 자체로 작업을 수행할 수 있다. 보다 일반적으로, 
+쿠버네티스에서는 컨트롤러가 
+{{< glossary_tooltip text="API 서버" term_id="kube-apiserver" >}} 로
+유용한 부수적인 효과가 있는 메시지를 발송한다. 사용자는 이 예시를 아래에서 볼수 있다.
+
+{{< comment >}}
+네임스페이스 컨트롤러와 같은 일부 내장된 컨트롤러는 사양을 가지지 않는 
+오브젝트에 대해 작동한다. 이에 대해 내용을 간단히 하기 위해서 이 페이지에서는 
+자세한 설명을 생략한다.
+{{< /comment >}}
+
+### API 서버를 통한 제어
+
+{{< glossary_tooltip term_id="job" >}} 컨트롤러는 쿠버네티스 
+내장 컨트롤러의 예시이다. 내장 컨트롤러는 클러스터 API서버와 
+상호 작용하며 상태를 관리한다.
+
+잡은 단일 {{< glossary_tooltip term_id="pod" >}} 또는 여러 파드를 실행하고, 
+작업을 수행한 다음 중지하는 
+쿠버네티스 리소스 이다.
+
+(일단 [스케줄되면](/ko/docs/concepts/scheduling/), 파드 오브젝트는 kubelet 
+의 의도한 상태 중 일부가 된다.)
+
+잡 컨트롤러가 새 작업을 볼 때, 클러스터 어딘가에서 
+노드 셋의 kubelet이 작업을 수행하기 위해 올바른 
+수의 파드를 실행하고 있는지 확인한다.
+잡 컨트롤러는 어떤 파드 또는 컨테이너를 스스로 실행하지 않는다.
+대신, 잡 컨트롤러는 API 서버에 파드를 생성하거나 삭제하도록 
+지시한다.
+{{< glossary_tooltip text="컨트롤 플레인" term_id="control-plane" >}}의 
+다른 컴포넌트는 새 정보
+(예약 및 실행해야 하는 새 파드)에 작용되며, 
+결국 작업이 완료된다.
+
+새 잡을 생성한 이후에는, 의도한 상태는 해당 잡을 완료하는 것이다.
+잡 컨트롤러는 현재 상태를 의도한 상태에 가깝게 
+만들며, 사용자가 원하는 잡을 수행하기위해 파드를 생성해서 
+잡이 완료에 가까워 지도록 한다.
+
+또한, 컨트롤러는 오브젝트의 설정을 업데이트 한다.
+예시: 한 작업이 종료된 잡은 잡 컨트롤러가 
+잡 오브젝트를 업데이트해서 `Finished` 로 표시한다.
+
+(이것은 지금 방 온도가 설정한 온도인 것을 표시하기 
+위해 실내 온도 조절기의 빛을 끄는 것과 약간 비슷하다).
+
+### 직접 제어
+
+잡과는 대조적으로, 일부 컨트롤러는 클러스터 외부의 것을 
+변경 해야 할 필요가 있다.
+
+예를 들어, 만약 컨트롤 루프를 사용해서 
+클러스터에 충분한 {{< glossary_tooltip text="노드들" term_id="node" >}}이
+있는지 확인하는 경우, 해당 컨트롤러는 필요할때 새 노드를 설정할 
+현재 클러스터 외부의 어떤 무언가를 필요로 한다.
+
+외부 상태와 상호 작용하는 컨트롤러는 API 서버에서 의도한
+상태를 찾은 다음, 외부 시스템과 직접 통신해서
+현재 상태를 보다 가깝게 만든다.
+
+(실제로 클러스터의 노드를 수평으로 확장하는
+컨트롤러가 있다.
+[클러스터 오토스케일링](/ko/docs/tasks/administer-cluster/cluster-management/#클러스터-오토스케일링) 을 본다.)
+
+## 원하는 상태와 현재 상태 {#desired-vs-current}
```

Comment:
You might want to leave the fragment identifiers as Korean text. Up to you @ysyukr.

Reply:
Okay.
But, this part may remain in other documents that are initially translated into Korean.
Since it was translated according to precedent, I hope that it will be decided by listening to the opinions of Korean team reviewers.

Reply:
In the other document I'm talking about, look at the following example.
example) /ko/docs/concepts/architecture/node.md
```markdown
### 주소 {#addresses}

이 필드의 용법은 클라우드 제공사업자 또는 베어메탈 구성에 따라 다양하다.
```
If I had to translate all of this, what would I do about the existing translated documents?

Reply:
@ysyukr I got your point. I checked similar cases for existing translated documents (there are 8 Ko docs) and I noticed that we didn't translate the fragment identifiers. Whereas there is no firm consensus on this yet, I hope not to translate the fragment identifiers for this moment for the consistency. 
@gochist Do you have any idea about this?

Reply:
I prefer not to override the fragment identifier. Because it means that the part of url is also translated. But as @seokho-son said, there's no consensus on it yet. So, for this PR, it is up to you, @ysyukr .

Reply:
@gochist @seokho-son 
There is no discussion of fragment identifiers, but I think it is best not to follow the precedents of translated documents first.

Reply:
@ysyukr Let me approve this PR. Hopefully, we would discuss (at least mention) about this translation issue in the coming Korean l10n meeting :)